projects/dendrites/permutedMNIST/run.py

```diff
@@ -0,0 +1,186 @@
+            # be in the range [0, 1, ..., 9]
+            target = target % exp.num_classes_per_task
+
+            data = data.to(exp.device)
+            context = context.to(exp.device)
+            target = target.to(exp.device)
+
+            output = exp.model(data, context)
+
+            # All output units are used to compute loss / accuracy
+            loss += exp.error_loss(output, target, reduction="sum")
+            pred = output.max(1, keepdim=True)[1]
+            correct += pred.eq(target.view_as(pred)).sum()
+            total += len(data)
+
+    mean_acc = torch.true_divide(correct, total).item() if total > 0 else 0,
+    return mean_acc
+
+
+def run_experiment(config):
+    exp_class = config["experiment_class"]
+    exp = exp_class()
+    exp.setup_experiment(config)
+
+    exp.model = exp.model.to(exp.device)
+
+    # Read optimizer class and args from config as it will be used to reinitialize the
+    # model's optimizer
+    optimizer_class = config.get("optimizer_class", torch.optim.SGD)
+    optimizer_args = config.get("optimizer_args", {})
+    num_tasks = config.get("num_tasks", 2)
+
+    # --------------------------- CONTINUAL LEARNING PHASE -------------------------- #
+
+    for task_id in range(num_tasks):
+
+        # Train model on current task
+        exp.train_loader.sampler.set_active_tasks(task_id)
+        for _epoch_id in range(exp.epochs):
+            train_model(exp)
+
+        if task_id in config["epochs_to_validate"]:
+
+            print("")
+            print(f"=== AFTER TASK {task_id} ===")
+            print("")
+
+            # Evaluate model accuracy on each task separately
+            for eval_task_id in range(task_id + 1):
+
+                exp.val_loader.sampler.set_active_tasks(eval_task_id)
+                acc_task = evaluate_model(exp)
+                if isinstance(acc_task, tuple):
+                    acc_task = acc_task[0]
+
+                print(f"task {eval_task_id} accuracy: {acc_task}")
+            print("")
+
+        else:
+            print(f"--Completed task {task_id}--")
+
+        # Reset optimizer before starting new task
+        del exp.optimizer
+        exp.optimizer = optimizer_class(exp.model.parameters(), **optimizer_args)
+
+    # ------------------------------------------------------------------------------- #
+
+    # Report final aggregate accuracy
+    exp.val_loader.sampler.set_active_tasks(range(num_tasks))
+    acc_task = evaluate_model(exp)
+    if isinstance(acc_task, tuple):
+        acc_task = acc_task[0]
+
+    print(f"Final test accuracy: {acc_task}")
+    print("")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        parents=DEFAULT_PARSERS,
+    )
+    parser.add_argument("-e", "--experiment", dest="name", default="default_base",
+                        help="Experiment to run", choices=CONFIGS.keys())
+    args = parser.parse_args()
```

Comment:
I noticed that `num_tasks` is hardcoded in the config (in base.py). Is the user able to specify `num_tasks` from the command line?

Reply:
It was hardcoded before too. Unfortunately it's not trivial to specify it on the command line because there are a number of other parameters in the config that depend on it (such as `num_segments`). Maybe we can address this in the next update.